src/Sulu/Bundle/MediaBundle/Resources/js/stores/CollectionStore/CollectionStore.js

```diff
@@ -1,84 +1,57 @@
 // @flow
-import {action, autorun, computed, observable} from 'mobx';
+import {computed} from 'mobx';
 import type {IObservableValue} from 'mobx'; // eslint-disable-line import/named
-import {ResourceRequester} from 'sulu-admin-bundle/services';
-import type {BreadcrumbItem, BreadcrumbItems, Collection} from './types';
+import {ResourceStore} from 'sulu-admin-bundle/stores';
 
 const COLLECTIONS_RESOURCE_KEY = 'collections';
 
 export default class CollectionStore {
-    @observable loading: boolean = false;
-    @observable collection: Collection = {
-        id: null,
-        parentId: null,
-        breadcrumb: null,
-    };
-    disposer: () => void;
+    collectionId: ?string | number;
+    locale: IObservableValue<string>;
+    resourceStore: ResourceStore;
 
     constructor(collectionId: ?string | number, locale: IObservableValue<string>) {
-        this.disposer = autorun(() => {
-            this.load(collectionId, locale.get());
-        });
+        this.collectionId = collectionId;
+        this.locale = locale;
+        this.resourceStore = new ResourceStore(
+            COLLECTIONS_RESOURCE_KEY,
+            collectionId,
+            {
+                locale,
+            },
+            {
+                depth: 1,
+                breadcrumb: true,
+                parent: true,
+            }
+        );
     }
 
     destroy() {
-        this.disposer();
-    }
-
-    @computed get id(): ?number {
-        return this.collection.id;
-    }
-
-    @computed get parentId(): ?number {
-        return this.collection.parentId;
+        this.resourceStore.destroy();
     }
 
-    @computed get breadcrumb(): ?BreadcrumbItems {
-        return this.collection.breadcrumb;
+    @computed get loading(): boolean {
+        return this.resourceStore ? this.resourceStore.loading : false;
     }
 
-    @action setLoading(loading: boolean) {
-        this.loading = loading;
+    @computed get id(): ?string | number {
+        return this.resourceStore.id;
     }
 
-    @action load(collectionId: ?string | number, locale: string) {
-        if (!collectionId) {
-            this.collection.breadcrumb = null;
+    @computed get parentId(): ?number {
```

Comment:
Is that the only remainer for what you need the `CollectionStore`?

Reply:
I think so but the `resourceStore` is also used through this store at some places